src/v3/src/components/LaunchAuthenticatorButton/LaunchAuthenticatorButton.tsx

```diff
@@ -30,10 +30,14 @@ const LaunchAuthenticatorButton: UISchemaElementComponent<{
     translations = [],
     options: {
       step,
+      deviceChallengeUrl,
     },
   } = uischema;
 
   const handleClick: ClickHandler = async () => {
+    if (deviceChallengeUrl) {
```

Comment:
Check out `src/util/Util.js#redirect` which is called in `SignInWithDeviceOption`. There's some special logic depending on if it's app link or not. I think we need to implement that here to ensure the right behavior.

https://github.com/okta/okta-signin-widget/blob/6f65f0b83c3cbd38acf817fd5bdbfa52b5e176e0/src/util/Util.js#L178-L188

Reply:
I'm in favor of copying the logic over rather than rely on the `Util.redirect` wrapper since we're going to eventually drop that anyways, but I know we already do use the `Util.redirectWithFormGet` in the v3 widget so we can continue to use that function for now.